monkey/monkey_island/cc/repositories/plugin_utils.py

```diff
@@ -0,0 +1,73 @@
+import json
+from tarfile import TarFile
+from typing import Any, BinaryIO, Dict
+
+import yaml
+
+from common.agent_plugins import AgentPlugin, AgentPluginManifest
+
+from . import RetrievalError
+
+MANIFEST_FILENAME = "plugin.yaml"
+CONFIG_SCHEMA_FILENAME = "config-schema.json"
+SOURCE_ARCHIVE_FILENAME = "data.tar"
+
+
+def load_plugin(file: BinaryIO) -> AgentPlugin:
```

Comment:
This could perhaps be a static method on `AgentPlugin`. Basically, an alternate constructor. Then this functionality would be available to other components besides repositories. Just throwing this out as an option.